lib/mcp_client/server_sse.rb

```diff
@@ -269,60 +262,31 @@ def start_sse_thread
       return if @sse_thread&.alive?
 
       @sse_thread = Thread.new do
-        sse_http = nil
-        begin
-          uri = URI.parse(@base_url)
-          sse_http = Net::HTTP.new(uri.host, uri.port)
-
-          if uri.scheme == 'https'
-            sse_http.use_ssl = true
-            sse_http.verify_mode = OpenSSL::SSL::VERIFY_PEER
-          end
-
-          sse_http.open_timeout = 10
-          sse_http.read_timeout = @read_timeout
-          sse_http.keep_alive_timeout = 60
-
-          sse_http.start do |http|
-            request = Net::HTTP::Get.new(uri)
-            @headers.each { |k, v| request[k] = v }
-
-            http.request(request) do |response|
-              unless response.is_a?(Net::HTTPSuccess) && response['content-type']&.start_with?('text/event-stream')
-                @mutex.synchronize do
-                  # Signal connection attempt completed (failed)
-                  @connection_established = false
-                  @connection_cv.broadcast
-                end
-                raise MCPClient::Errors::ServerError, 'Server response not OK or not text/event-stream'
-              end
+        uri = URI.parse(@base_url)
+        sse_base = "#{uri.scheme}://#{uri.host}:#{uri.port}"
+        sse_path = uri.request_uri
 
-              @mutex.synchronize do
-                # Signal connection established and SSE ready
-                @sse_connected = true
-                @connection_established = true
-                @connection_cv.broadcast
-              end
+        @sse_conn ||= Faraday.new(url: sse_base) do |f|
```

Comment:
why are you reopening a new connection, you will need to redo the session initialization .